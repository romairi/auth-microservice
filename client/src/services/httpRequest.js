// import axios from "axios";
// import qs from "qs";

// import { getSessionToken } from "./utilities/globalUtils/utils";
// import envDetails from "./env.json";

// const serverUrl = envDetails[process.env.NODE_ENV].serverUrl + "/api";

// const onSuccess = ({ data }, resolve, reject) => {
//   if (data.status !== "success") {
//     return reject(data);
//   }

//   resolve(data);
// };

// const onFailure = (err, reject) => {
//   console.error(err);
//   reject({ description: "Network error" });
// };

// const getOptions = () => ({
//   headers: {
//     "Content-Type": "application/json",
//     Authorization: getSessionToken(),
//   },
// });

// const post = (path, data = {}) =>
//   new Promise((resolve, reject) => {
//     axios
//       .post(serverUrl + path, data, getOptions())
//       .then((res) => onSuccess(res, resolve, reject))
//       .catch((err) => onFailure(err, reject));
//   });

// const get = (path, params) =>
//   new Promise((resolve, reject) => {
//     const pathWithParams =
//       serverUrl + path + (params ? `?${qs.stringify(params)}` : "");

//     axios
//       .get(pathWithParams, getOptions())
//       .then((res) => onSuccess(res, resolve, reject))
//       .catch((err) => onFailure(err, reject));
//   });

// const put = (path, data = {}) =>
//   new Promise((resolve, reject) => {
//     axios
//       .put(serverUrl + path, data, getOptions())
//       .then((res) => onSuccess(res, resolve, reject))
//       .catch((err) => onFailure(err, reject));
//   });

// const del = (path, params) =>
//   new Promise((resolve, reject) => {
//     const pathWithParams =
//       serverUrl + path + (params ? `?${qs.stringify(params)}` : "");

//     axios
//       .delete(pathWithParams, getOptions())
//       .then((res) => onSuccess(res, resolve, reject))
//       .catch((err) => onFailure(err, reject));
//   });

// const httpRequest = {
//   post,
//   get,
//   put,
//   del,
// };

// export default httpRequest;
