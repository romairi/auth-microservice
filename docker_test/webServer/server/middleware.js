import express from "express";
import path from "path";
import userRoutes from "./user/routes";
import {
  logErrors,
  clientErrorHandler,
  errorHandler,
} from "./services/errorHandling";
import { serverConfig } from "./config/serverConfig";

/**
 * Application-Level Middleware
 * @param app
 * @param logger
 */
export const middleware = (app) => {
  app.use(express.json());
  app.use(
    "/public",
    express.static(path.join(__dirname, serverConfig.staticFolder))
  );
  app.use("/auth", userRoutes);
  app.get("/*", (req, res) => {
    res.sendFile(path.join(__dirname, serverConfig.staticFolder, "index.html"));
  });
  app.use(logErrors(console));
  app.use(clientErrorHandler);
  app.use(errorHandler);
};
