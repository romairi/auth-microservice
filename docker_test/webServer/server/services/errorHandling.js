import { StatusCodes, ReasonPhrases } from "http-status-codes";

/**
 * This is generic function writes request and error information to stderr
 * @param logger
 * @returns {Function}
 */
export const logErrors = (logger) => (err, req, res, next) => {
  logger.error(err.stack);
  next(err);
};

/**
 * The error is explicitly passed along to the next one.
 * @param err
 * @param req
 * @param res
 * @param next
 */
export function clientErrorHandler(err, req, res, next) {
  if (req.xhr) {
    res
      .status(StatusCodes.INTERNAL_SERVER_ERROR)
      .send({ error: ReasonPhrases.INTERNAL_SERVER_ERROR });
  } else {
    next(err);
  }
}

/**
 * This function catches all errorHandlers
 * @param err
 * @param req
 * @param res
 * @param next
 */
export function errorHandler(err, req, res, next) {
  res.status(StatusCodes.INTERNAL_SERVER_ERROR);
  res.send(ReasonPhrases.INTERNAL_SERVER_ERROR);
}
