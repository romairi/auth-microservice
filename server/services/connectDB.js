import mongoose from "mongoose";
import {serverConfig} from "../config/serverConfig";
import Logger from "./logger";

/**
 * Connection to MongoDB with specifying several more parameters in the URI
 */
export const connectDB = (logger = new Logger(), {process: _process} = globalThis) => {
    return mongoose.connect(serverConfig.mongo.hostURI, {
        useNewUrlParser: true,
        useUnifiedTopology: true,
        useFindAndModify: false,
        useCreateIndex: true
    })
        .then(() => logger.log('MongoDB Connected...'))
        .catch(err => {
            logger.log("MONGO ERROR");
            logger.error(err.message);
            _process.exit(1);
        })
};
