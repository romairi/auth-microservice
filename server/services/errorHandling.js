import {StatusCodes, ReasonPhrases} from 'http-status-codes';
import {ValidationError} from "express-validation";


/**
 * An object details keyed by parameter, each containing an array of errors in joi format
 * @param err
 * @param req
 * @param res
 * @param next
 * @returns by default will return errors in the following format,
 */
export function validationError(err, req, res, next) {
    if (err instanceof ValidationError) {
        return res.status(err.statusCode).json(err);
    }
    return next(err);
}

/**
 * This is generic function writes request and error information to stderr
 * @param err
 * @param req
 * @param res
 * @param next
 */
export function logErrors(err, req, res, next) {
    console.error(err.stack);
    next(err);
}

/**
 * The error is explicitly passed along to the next one.
 * @param err
 * @param req
 * @param res
 * @param next
 */
export function clientErrorHandler(err, req, res, next) {
    if (req.xhr) {
        res.status(StatusCodes.INTERNAL_SERVER_ERROR).send({error: ReasonPhrases.INTERNAL_SERVER_ERROR});
    } else {
        next(err);
    }
}

/**
 * This function catches all errorHandlers
 * @param err
 * @param req
 * @param res
 * @param next
 */
export function errorHandler(err, req, res, next) {
    res.status(StatusCodes.INTERNAL_SERVER_ERROR);
    res.send(ReasonPhrases.INTERNAL_SERVER_ERROR);
}
