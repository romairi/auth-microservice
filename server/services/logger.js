/**
 * A wrapper logger
 */
export default class Logger {
  constructor(_console = globalThis.console) {
    this.console = _console;
  }

  log(...args) {
    this.console.log(...args);
  }

  info(...args) {
    this.console.info(...args);
  }

  error(...args) {
    this.console.error(...args);
  }
}
