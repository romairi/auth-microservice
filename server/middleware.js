import express from 'express';
import userRoutes from './user/routes';
import {validationError, logErrors, clientErrorHandler, errorHandler} from './services/errorHandling';

/**
 * Application-Level Middleware
 * @param app
 */
export const middleware = app => {
    app.use(express.json());

    app.use('/', userRoutes);
    app.use(validationError);
    app.use(logErrors);
    app.use(clientErrorHandler);
    app.use(errorHandler);
};
