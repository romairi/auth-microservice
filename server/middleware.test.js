import { logErrors } from "./services/errorHandling";
import { middleware } from "./middleware";

jest.mock("express", () => ({ json: () => "EXPRESS_JSON" }));
jest.mock("./user/routes", () => "USER_ROUTE");
jest.mock("./services/errorHandling", () => ({
  logErrors: jest.fn(() => "LOG_ERRORS"),
  validationError: "VALIDATION_ERROR",
  clientErrorHandler: "CLIENT_ERROR_HANDLER",
  errorHandler: "ERROR_HANDLER",
}));

describe("middleware", () => {
  const appMock = { use: jest.fn() };
  const loggerMock = {};

  it("should call app.use with the middleware", () => {
    middleware(appMock, loggerMock);

    expect(logErrors).toHaveBeenCalledWith(loggerMock);
    expect(appMock.use.mock.calls).toMatchSnapshot();
  });
});
