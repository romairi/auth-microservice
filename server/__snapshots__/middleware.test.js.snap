// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`middleware should call app.use with the middleware 1`] = `
Array [
  Array [
    "EXPRESS_JSON",
  ],
  Array [
    "/",
    "USER_ROUTE",
  ],
  Array [
    "VALIDATION_ERROR",
  ],
  Array [
    "LOG_ERRORS",
  ],
  Array [
    "CLIENT_ERROR_HANDLER",
  ],
  Array [
    "ERROR_HANDLER",
  ],
]
`;
